### PR TITLE
Update DEPS to pull in libtess2, sqlite, and, inja.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Also updates the license script to appropriately handle these
dependencies.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
@Hixie was taking over progress on this but ran into roadblocks. @Hixie were the roadblocks related to the new dependencies being added? If not, may I add support in the script just for these dependencies?

**Comment by Hixie** [comment]
Sorry, I got distracted by other unrelated issues. To fix the script for libtess2, just add the following to `tools/licenses/lib/patterns.dart`:

```patch
@@ -1573,20 +1593,27 @@ final List<RegExp> csLicenses = <RegExp>[
       r'|'

       r'\1\2 '
       r'The above copyright notice and this permission notice'
       r'(?: \(including the next paragraph\))? '
       r'shall be included in all copies or substantial portions '
       r'of the (?:Software|Materials)\.'

       r'|'

+      r'\1\2 '
+      r'The above copyright notice including the dates of first publication and either this '
+      r'permission notice or a reference to .+ shall be '
+      r'included in all copies or substantial portions of the Software.'
+
+      r'|'
+
       r'\1\2 '
       r'In addition, the following condition applies:'

       r'|'

       r'\1\2 '
       r'All redistributions must retain an intact copy of this copyright notice and disclaimer\.'

       r'|'

@@ -1606,20 +1633,27 @@ final List<RegExp> csLicenses = <RegExp>[
       r'OR FITNESS FOR A PARTICULAR PURPOSE\.'

       r'|'

       r'\1\2 '
       r'IN NO EVENT SHALL .+ BE LIABLE FOR ANY SPECIAL, INCIDENTAL, INDIRECT OR CONSEQUENTIAL '
       r'DAMAGES OF ANY KIND, OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, '
       r'WHETHER OR NOT ADVISED OF THE POSSIBILITY OF DAMAGE, AND ON ANY THEORY OF LIABILITY, ARISING '
       r'OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE\.'

+      r'|'
+
+      r'\1\2 '
+      r'Except as contained in this notice, the name of .+ shall not '
+      r'be used in advertising or otherwise to promote the sale, use or other dealings in '
+      r'this Software without prior written authorization from .+\.'
+
       .replaceAll(' ', _linebreak)
     )
     +
     r')*',
     multiLine: true,
     caseSensitive: false
   ),

   // BISON LICENSE
   // Seen in some ANGLE source. The usage falls under the "special exception" clause.
```

To skip INJA, add the following to `_RepositoryRootThirdPartyDirectory.shouldRecurse` in the appropriate place:

```patch
+        && entry.name != 'inja' // build-time only according to chinmay
```

That should get you far enough to add the dependencies without the CI failing. I'll submit a follow-up patch shortly that cleans up some of the issues I ran into. I don't think they're actually causing us to show the wrong licenses; I think they're causing us to show the right licenses but for the wrong reasons.

**Comment by chinmaygarde** [comment]
Thanks for the help! The patch should be good for a review now.

File: tools/licenses/lib/main.dart [hunk 1]
```diff
@@ -1850,7 +1850,9 @@ class _RepositoryRootThirdPartyDirectory extends _RepositoryGenericThirdPartyDir
         && entry.name != 'spirv_headers' // only used on hosts for tests
         && entry.name != 'spirv_cross' // only used on hosts for tests
         && entry.name != 'ocmock' // only used for tests
-        && entry.name != 'java'// only used for Android builds
+        && entry.name != 'java' // only used for Android builds
+        && entry.name != 'java' // only used for Android builds
```
Comment (zanderso) [inline tools/licenses/lib/main.dart:1854]: Looks like this line is duplicated.

**Review by zanderso — APPROVED** [review]
lgtm w/ nit

**Comment by Hixie** [comment]
Don't forget to also update the actual LICENSE file, that's the one that matters the most.

**Comment by chinmaygarde** [comment]
Done.

**Pull request merged**